File: 121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        profit = 0
        min_amt = 9999999999999

        for i in prices:
            if i < min_amt:
                min_amt = i
            elif (i - min_amt) > profit:
                profit = i - min_amt


        return profit
        

File: C-Sharp Solutions/Sliding Window/121. Best Time to Buy and Sell Stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int max_profit = 0;
        int left = 0;

        for (int right = 0; right < prices.Length; right++) {
            if (right == left) {
                continue;
            }

            if (prices[right] > prices[left]) {
                var temp_prof = prices[right] - prices[left];

                if (temp_prof > max_profit) {
                    max_profit = temp_prof;
                }
            } else if (prices[right] < prices[left]) {
                left = right;
            }

        }

        return max_profit;
    }
}
